martha/martha_v3.js

```diff
@@ -471,30 +290,27 @@ function validateRequest(url, auth, requestedFields) {
     }
 }
 
-/**
- * Used to check if any of the requested fields overlap with fields dependent on an underlying service.
- *
- * @param {string[]} requestedFields
- * @param {string[]} serviceFields
- * @returns {boolean} true if the requested fields overlap
- */
-function overlapFields(requestedFields, serviceFields) {
-    // via https://medium.com/@alvaro.saburido/set-theory-for-arrays-in-es6-eb2f20a61848
-    return requestedFields.filter((field) => serviceFields.includes(field)).length !== 0;
-}
-
 function buildRequestInfo(params) {
     const {
         url,
         requestedFields,
         auth,
+        forceAccessUrl,
     } = params;
 
     validateRequest(url, auth, requestedFields);
-    const drsType = determineDrsType(url);
+    const urlParts = getHttpsUrlParts(url);
+
+    // Force the retrieval of a (signed) access URL for this request if the `martha-force-access-url` header is set
+    // copying the original object with `forceAccessUrl` set appropriately.
+    // Shallow clone a JavaScript ES6 class instance
+    // https://stackoverflow.com/questions/41474986/how-to-clone-a-javascript-es6-class-instance
+    const drsProvider = { ...determineDrsProvider(url, urlParts), forceAccessUrl: Boolean(forceAccessUrl) };
+    Object.setPrototypeOf(drsProvider, DrsProvider.prototype);
```

Comment:
I'm curious why not just have `determineDrsProvider` return a new instance every time, but this should also work.